### PR TITLE
refactor: include batch lockup contract in core

### PR DESCRIPTION
**Pull request by andreivladbrg**
Depends on #1075 Closes https://github.com/sablier-labs/v2-core/issues/1073

### CODE REVIEW

**Comment by andreivladbrg** [comment]
@smol-ninja the PR is ready for review, maybe partially reviewing it while working on https://github.com/sablier-labs/v2-core/issues/1072 would be useful

**Comment by smol-ninja** [comment]
Since its built on top of https://github.com/sablier-labs/v2-core/pull/1075, I would like to give it a review after https://github.com/sablier-labs/v2-core/pull/1075 is merged.

File: script/core/DeployDeterministicBatchLockup.t.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+pragma solidity >=0.8.22 <0.9.0;
+
+import { SablierBatchLockup } from "../../src/core/SablierBatchLockup.sol";
+
+import { BaseScript } from "../Base.s.sol";
+
+contract DeployDeterministicBatchLockup is BaseScript {
+    /// @dev Deploy via Forge.
+    function run() public virtual broadcast returns (SablierBatchLockup batchLockup) {
+        batchLockup = new SablierBatchLockup();
```
Comment (smol-ninja) [inline script/core/DeployDeterministicBatchLockup.t.sol]: missing `SALT`

File: script/periphery/DeployDeterministicMerkleFactory.s.sol
Comment (smol-ninja) [inline script/periphery/DeployDeterministicMerkleFactory.s.sol:1]: Not required as will be deleted from this repo

**Review by smol-ninja — APPROVED** [review]
LGTM. 

Added a small commit: https://github.com/sablier-labs/v2-core/pull/1084/commits/45b1555f72637c687c0c8125247a7c27219ee59d

File: script/core/DeployDeterministicBatchLockup.t.sol (hunk 1 above)
Comment (smol-ninja) [reply script/core/DeployDeterministicBatchLockup.t.sol]: https://github.com/sablier-labs/v2-core/pull/1084/commits/45b1555f72637c687c0c8125247a7c27219ee59d

Comment (andreivladbrg) [reply script/core/DeployDeterministicBatchLockup.t.sol]: thx

**Pull request merged**